explorer/ast/declaration.h

```diff
@@ -126,6 +126,59 @@ class Declaration : public AstNode {
   bool is_type_checked_ = false;
 };
 
+// A name being declared in a named declaration.
+class DeclaredName {
+ public:
+  struct NameComponent {
+    SourceLocation source_loc;
+    std::string name;
+  };
+
+  // This is required by bison, and should not be used elsewhere.
+  explicit DeclaredName() {}
+
+  explicit DeclaredName(SourceLocation loc, std::string name)
+      : components_{{loc, name}} {}
```

Comment:
```suggestion
      : components_{{loc, std::move(name)}} {}
```

Maybe run tidy? I think it might've caught this?